modules/javafx.controls/src/test/java/test/javafx/scene/control/ListCellTest.java

```diff
@@ -833,5 +836,88 @@ public void testListCellHeights() {
+        assertChangeIndexToEditing(0, 1);
+    }
+
+    @Test
+    public void testChangeIndexToEditing2_jdk_8264127() {
+        assertChangeIndexToEditing(1, 0);
+    }
+
+    @Test
+    public void testChangeIndexToEditing3_jdk_8264127() {
+        assertChangeIndexToEditing(1, -1);
+    }
+
+    @Test
+    public void testChangeIndexToEditing4_jdk_8264127() {
+        assertChangeIndexToEditing(0, -1);
+    }
+
+    @Test
+    public void testChangeIndexToEditing5_jdk_8264127() {
+        assertChangeIndexToEditing(-1, 0);
+    }
+
+    private void assertChangeIndexToEditing(int initialCellIndex, int listEditingIndex) {
+        list.getFocusModel().focus(-1);
+        List<EditEvent> events = new ArrayList<EditEvent>();
+        list.setOnEditStart(e -> {
+            events.add(e);
+        });
+        list.setEditable(true);
+        cell.updateListView(list);
+        cell.updateIndex(initialCellIndex);
+        list.edit(listEditingIndex);
+        assertEquals("sanity: list editingIndex ", listEditingIndex, list.getEditingIndex());
+        assertFalse("sanity: cell must not be editing", cell.isEditing());
+        cell.updateIndex(listEditingIndex);
+        assertEquals("sanity: index updated ", listEditingIndex, cell.getIndex());
+        assertEquals("list editingIndex unchanged by cell", listEditingIndex, list.getEditingIndex());
+        if(listEditingIndex != -1) {
+            assertTrue(cell.isEditing());
+            assertEquals(1, events.size());
+        } else {
+            // -1 represents "not editing" for the listview and "no index" for the list cell.
+            assertFalse(cell.isEditing());
+            assertEquals(0, events.size());
+        }
+    }
+
+    @Test
+    public void testChangeIndexOffEditing0_jdk_8264127() {
+        assertUpdateCellIndexOffEditing(1, 0);
+    }
+    @Test
+    public void testChangeIndexOffEditing1_jdk_8264127() {
+        assertUpdateCellIndexOffEditing(1, -1);
+    }
+    @Test
+    public void testChangeIndexOffEditing2_jdk_8264127() {
+        assertUpdateCellIndexOffEditing(0, 1);
+    }
+    @Test
+    public void testChangeIndexOffEditing3_jdk_8264127() {
+        assertUpdateCellIndexOffEditing(0, -1);
+    }
+
+    public void assertUpdateCellIndexOffEditing(int editingIndex, int cancelIndex) {
+        list.getFocusModel().focus(-1);
+        List<EditEvent> events = new ArrayList<EditEvent>();
+        list.setOnEditCancel(e -> {
+            events.add(e);
+        });
+        list.setEditable(true);
+        cell.updateListView(list);
+        cell.updateIndex(editingIndex);
+        list.edit(editingIndex);
+        // here we are certain that the cell is in editing state
+        assertTrue("sanity: cell must be editing", cell.isEditing());
+        cell.updateIndex(cancelIndex); // change cell index to negative
+        assertFalse("cell must not be editing if cell index is " + cell.getIndex(), cell.isEditing());
+        assertEquals(1, events.size());
+    }
```

Comment:
still missing: assert that list editingIndex is unchanged (see the assertTo test). Without that assert, you don't see any difference in the test between your previous and this commit (which was surrounding the call to cancelEdit with setting the updateEditingIndex flags), do you ;) 

A personal note: tests are our friends, red-green (repeatedly) the color sequence that brings satisfaction :) Just changing a code snippet without making certain to have a failing test before (and passes after) a change is .. well .. suboptimal.

Reply:
Yes, i can confirm that it's required to find the issue with the updateEditingIndex. I've added more tests and in general its now more symmetric to the other test!

Reply:
missing test for the try/finally